src/openassetio-python/CMakeLists.txt

```diff
@@ -99,28 +99,63 @@ if (OPENASSETIO_ENABLE_TESTS AND Python_Interpreter_FOUND)
         COMMAND
         "${Python_EXECUTABLE}" -m venv "${CMAKE_INSTALL_PREFIX}"
         COMMAND
-        ${venv_python_dir}/python -m pip install --upgrade pip
+        ${venv_python_dir}/python -m pip install --upgrade --force-reinstall pip
```

Comment:
[plan] This fails if I `clean` then disconnect from the internet then attempt to build. 

Arguably that's to be expected, and regardless is probably out of scope for this PR, so we should make a separate issue.

I think CMake `clean` deleting (only) the pip binary is just wrong and we should ultimately find a better solution. When I originally wrote this CMake I didn't consider the effect of `clean` on the `OUTPUT` of an `add_custom_command`, so my bad.

I'll make an issue.

Reply:
Added #311